cmd/crio/main.go

```diff
@@ -347,6 +368,22 @@ func main() {
 		<-serverCloseCh
 		logrus.Debugf("Closed main server")
 
+		memProfilePath := c.String("profile-mem")
+		if memProfilePath != "" {
+			logrus.Infof("Creating memory profile in: %v", memProfilePath)
+
+			file, err := os.Create(memProfilePath)
+			if err != nil {
+				return errors.Wrap(err, "could not create memory profile")
+			}
+			defer file.Close()
+			runtime.GC()
```

Comment:
Why do we need this?

Reply:
To update the statistics of the profile.

For example, let's compare those three memory profiles. What I did is starting CRI-O and running `crictl version` and `crictl images` consecutively.

1. With the http pprof endpoint: `sudo curl --unix-socket /var/run/crio/crio.sock "http://localhost/debug/pprof/heap"`
![profile001](https://user-images.githubusercontent.com/695473/131967141-1ed1493e-28ac-46f7-b878-01ff2cb4a3c5.gif)

2. With the http pprof endpoint: `sudo curl --unix-socket /var/run/crio/crio.sock "http://localhost/debug/pprof/heap"` and running the GC with SIGUSR2: `sudo kill -SIGUSR2 $(pgrep crio)`
![profile002](https://user-images.githubusercontent.com/695473/131967231-938128ee-d5ca-4753-afc1-745d5be7e4b5.gif)

3. With the new `--profile-mem` option
![profile003](https://user-images.githubusercontent.com/695473/131967252-f820a769-7d8c-4a20-aa8d-0169bd54c90f.gif)

Reply:
I meant the call to runtime.GC() :)

Reply:
Oh it's part of the original example: https://pkg.go.dev/runtime/pprof